MSVC fails building due to `min` in core/array.h:93

**Godot version:**
c6e3394

**OS/device including version:**
windows 7 / MSBuild 2017


**Issue description:**
Build is called with this command (this was working yesterday):

    call scons.bat platform=windows module_mono_enabled=yes tools=yes  mono_glue=no

Build fails with these errors:
````
repo-path\core\array.h(93): error C2059: syntax error: ')'
repo-path\core\array.h(93): error C2334: unexpected token(s) preceding ':'; skipping apparent function body
repo-path\core\array.h(99): error C2143: syntax error: missing ')' before ';'
repo-path\core\array.h(99): error C2059: syntax error: ')'
repo-path\core\array.h(99): error C2238: unexpected token(s) preceding ';'
repo-path\core\cowdata.h(183): error C3254: 'Array': class contains explicit override '_unref' but does not derive from an interface that contains the function declaration
repo-path\core\cowdata.h(183): error C2838: '_unref': illegal qualified name in member declaration
repo-path\core\cowdata.h(207): error C3254: 'Array': class contains explicit override '_copy_on_write' but does not derive from an interface that contains the function declaration
repo-path\core\cowdata.h(207): error C2838: '_copy_on_write': illegal qualified name in member declaration
repo-path\core\cowdata.h(237): error C3254: 'Array': class contains explicit override 'resize' but does not derive from an interface that contains the function declaration
repo-path\core\cowdata.h(237): error C2838: 'resize': illegal qualified name in member declaration
repo-path\core\cowdata.h(305): error C3254: 'Array': class contains explicit override '_ref' but does not derive from an interface that contains the function declaration
repo-path\core\cowdata.h(305): error C2838: '_ref': illegal qualified name in member declaration
repo-path\core\cowdata.h(310): error C3254: 'Array': class contains explicit override '_ref' but does not derive from an interface that contains the function declaration
repo-path\core\cowdata.h(310): error C2838: '_ref': illegal qualified name in member declaration
repo-path\core\cowdata.h(327): error C3254: 'Array': class contains explicit override '{ctor}' but does not derive from an interface that contains the function declaration
repo-path\core\cowdata.h(327): error C2838: '{ctor}': illegal qualified name in member declaration
repo-path\core\cowdata.h(333): error C3254: 'Array': class contains explicit override '{dtor}' but does not derive from an interface that contains the function declaration
repo-path\core\cowdata.h(333): error C2838: '{dtor}': illegal qualified name in member declaration
repo-path\core\cowdata.h(333): error C2897: a destructor cannotbe a function template
repo-path\core\vector.h(154): error C3254: 'Array': class contains explicit override 'invert' but does not derive from an interface that contains the function declaration
repo-path\core\vector.h(154): error C2838: 'invert': illegal qualified name in member declaration
repo-path\core\vector.h(163): error C3254: 'Array': class contains explicit override 'append_array' but does not derive from an interface that contains the function declaration
repo-path\core\vector.h(163): error C2838: 'append_array': illegal qualified name in member declaration
repo-path\core\vector.h(174): error C3254: 'Array': class contains explicit override 'push_back' but does not derive from an interface that contains the function declaration
repo-path\core\vector.h(174): error C2838: 'push_back': illegalqualified name in member declaration
repo-path\core\ustring.h(302): error C2804: binary 'operator ==' has too many parameters
repo-path\core\ustring.h(304): error C2804: binary 'operator +' has too many parameters
repo-path\core\ustring.h(305): error C2804: binary 'operator +' has too many parameters
repo-path\core/print_string.h(36): error C2071: 'Array::_print_func': illegal storage class
repo-path\core/print_string.h(57): error C2071: 'Array::_print_line_enabled': illegal storage class
repo-path\core/print_string.h(58): error C2071: 'Array::_print_error_enabled': illegal storage class
repo-path\core/print_string.h(59): error C2720: 'Array::print_line': 'extern' storage-class specifier illegal on members
repo-path\core/print_string.h(60): error C2720: 'Array::print_error': 'extern' storage-class specifier illegal on members
repo-path\core/print_string.h(61): error C2720: 'Array::print_verbose': 'extern' storage-class specifier illegal on members
modules\mono\utils\thread_local.cpp(42): error C2911: 'ThreadLocalStorage::Impl': cannot be declared or defined in the current scope
modules\mono\utils\thread_local.cpp(90): error C3254: 'Array': class contains explicit override 'get_value' but does not derive from an interface that contains the function declaration
modules\mono\utils\thread_local.cpp(90): error C2838: 'get_value': illegal qualified name in member declaration
modules\mono\utils\thread_local.cpp(94): error C3254: 'Array': class contains explicit override 'set_value' but does not derive from an interface that contains the function declaration
modules\mono\utils\thread_local.cpp(94): error C2838: 'set_value': illegal qualified name in member declaration
modules\mono\utils\thread_local.cpp(98): error C3254: 'Array': class contains explicit override 'alloc' but does not derive from an interface that contains the function declaration
modules\mono\utils\thread_local.cpp(98): error C2838: 'alloc': illegal qualified name in member declaration
modules\mono\utils\thread_local.cpp(104): error C3254: 'Array': class contains explicit override 'free' but does not derive from an interface that contains the function declaration
modules\mono\utils\thread_local.cpp(104): error C2838: 'free': illegal qualified name in member declaration
modules\mono\utils\thread_local.cpp(108): fatal error C1004: unexpected end-of-file found
scons: *** [modules\mono\utils\thread_local.windows.tools.64.obj] Error 2
scons: building terminated because of errors.
````

--- a/platform/windows/detect.py
+++ b/platform/windows/detect.py
@@ -210,6 +210,7 @@ def configure_msvc(env, manual_msvc_config):
                                    'WIN32', 'MSVC',
                                    {'WINVER' : '$target_win_version',
                                     '_WIN32_WINNT': '$target_win_version'}])
+    env.AppendUnique(CPPDEFINES=['NOMINMAX']) # disable bogus min/max WinDef.h macros
     if env["bits"] == "64":
         env.AppendUnique(CPPDEFINES=['_WIN64'])
 

